### DISCUSSION TITLE
blog/fisher-s-exact-test-in-r-independence-test-for-a-small-sample/

### DISCUSSION THREAD

**Discussion by giscus** [AntoineSoetewey/statsandr#91] [General]
# blog/fisher-s-exact-test-in-r-independence-test-for-a-small-sample/

Learn the differences between the Chi-square test of independence and the Fisher's exact test. You will also learn how to implement the statistical test in R

https://statsandr.com/blog/fisher-s-exact-test-in-r-independence-test-for-a-small-sample/

**Comment by AekKaw** [comment]
Hello,
Many thanks for this very helpful post. I need help for my study.
I count, each month, the number of insects captured in 3 different orchards. The following contingency table summarizes the data:

```
Month		orchards1	 orchards2	orchards3
January		0		        0		            0
February	    0		        0		            0
March		    0		        0		            0
April		       3		       0		           1
May		     42		       1		          21
June		    260		      132		     242
July		    283		      40		          54
August		70		     197		         312
Septem 	567		      126		      180
October	481		       46		          110
Novem	     266		       6		            3
Decemb	 59		       0		            0
Total         2031          548              923
```

I would like to apply Fisher's exact test on rows to have pvalue for each row and group's letter in each cell as follow (pvalues & letter are not correct, just to be more clear on what i need as result)
```
Month		orchards1	 orchards2	orchards3    Pvalue
January		0a		    0a		  0a                    1
February	    0a	        0a          0a                    1
March		    0a	        0a          0a                    1
April		       3a		       0a		      1a                   1
May		     42a		       1b		      21a               0.001
June		    260a		   132b		 242b              0.0145
July		    283a	      40b          54b               0.0036
August		70		     197a	      312a               0.28
Septem 	567a	      126b	      180c              0.043
October	481a	       46a		   110a              0.31
Novem	     266a	       6b	            3b                0.0005
Decemb	 59a		       0b		         0b                0.0003
Total         2031a        548b          923 ab           0.002
```

Thanks in advance

**Reply by AntoineSoetewey** [reply]
Hello,

As for [Chi-square test for independence](https://statsandr.com/blog/chi-square-test-of-independence-in-r/), the exact Fisher test is used to test the association between ***two*** categorical variables.

If you want perform a test on each row, it means you only have ***one*** categorical variable (orchards).

With one categorical variable (with 3 levels like you have), you can do the [Chi-square goodness of fit test](https://statsandr.com/blog/one-proportion-and-goodness-of-fit-test-in-r-and-by-hand/#chi-square-goodness-of-fit-test) (to test whether the proportion is the same across all orchards for example).

Hope this helps.

Regards,
Antoine

**Comment by AekKaw** [comment]
Thanks for your quick reply.
I'll return to you if needed.

**Reply by AntoineSoetewey** [reply]
You're welcome!

**Comment by rhkoly** [comment]
Fisher's Exact Test 

data:  G6PDGenderTable
p-value = 0.0002968
alternative hypothesis: two.sided


after running the Fisher exact test R returned above value. How would I interpret weather the variables(G6PD and Gender )are independent or dependent?
TIA

**Reply by AntoineSoetewey** [reply]
The null and alternative hypotheses of the Fisher exact test are (see more in this [section](https://statsandr.com/blog/fisher-s-exact-test-in-r-independence-test-for-a-small-sample/#hypotheses)):

- H0: variables are independent
- H1: variables are dependent

Conclusion: we reject the null hypothesis of independence and conclude that G6PD and Gender are dependent (p-value < 0.001).

**Comment by drcox633** [comment]
Hi Antoine, 

Similar in a way to AekKaw's question. I have a 2x3 matrix, where I have two categorical variables one with a yes or no response and the other that breaks into three classified groups. 

example table:
History of fever        flu detected       other virus         none detected
                   Yes               12                       4                           8
                   No                7                       10                           4

I am just trying to understand how the fischer test functions in R works with a 2x3 matrix and if I need a separate p-value and test result for each group comparison. Such as flue detected vs other, other vs none, flu vs none. Or would a single p-value be sufficient? I feel like it wouldnt work for a single p-value since I wouldnt know how the groups compare and interact with each other more granularly than an overall the three groups are independent or dependent, correct?

Should I split my 2x3 into the three separate comparative cases and do the fischer test on each?
Thank you!

**Reply by AntoineSoetewey** [reply]
Hello,

First of all, whether you choose to do a single test on both variables, or 3 tests to compare Yes vs. No for each level (flu detected, other virus or none detected) depends on your research question:

- If you want to check whether the history of fever (Yes or No) is associated with the virus detected (flu, other or none), you should do a [Chi-square test of independence](https://statsandr.com/blog/chi-square-test-of-independence-in-r/) or a [Fisher's exact test](https://statsandr.com/blog/fisher-s-exact-test-in-r-independence-test-for-a-small-sample/). Chi-square test if all expected frequencies are above 5, otherwise Fisher's exact test. Note that with your data, all expected frequencies > 5. 
- If you are interested in comparing the proportions of Yes versus No **within each level of the variable virus**, then you should do a [one-proportion test](https://statsandr.com/blog/one-proportion-and-goodness-of-fit-test-in-r-and-by-hand/) (and not a Fisher's exact test as you mentioned because this test is used for **two** variables).

Now let me repeat what you wrote to make sure I understand well. You are interested in checking whether the two variables (history of fever and virus) are dependent or not. But if (based on the result of the independence test) you conclude that they are dependent, you also want more information than just concluding that they are dependent (you perhaps want to know which groups make the variables dependent and how they interact with each other).

This is possible with a mosaic plot. See more information in this [section](https://statsandr.com/blog/chi-square-test-of-independence-in-r/#combination-of-plot-and-statistical-test).

With your data, it would lead to this plot:

![image](https://user-images.githubusercontent.com/17910063/143778622-41c392ae-b108-4999-a4cd-7b02505b2ac2.png)

As you can see, we do no reject the null hypothesis that the history of fever and the virus are independent (*p*-value = 0.08).

But if the two variables were dependent (i.e., *p*-value < 0.05), the mosaic plot would have shown which subgroups make the variables dependent and how they interact with each other (see the example with the areas colored in the article).

An alternative is the following stacked barplot:

![image](https://user-images.githubusercontent.com/17910063/143778898-5305ac9c-e4bf-471d-ae49-141e61847e02.png)

The proportions Yes versus No are somewhat different for the 3 levels of virus (33% No when no virus is detected, whereas it is 71% when it is another virus), but this difference seems to be too small in the [samples](https://statsandr.com/blog/what-is-the-difference-between-population-and-sample/) to conclude that there is significant association between the two variables in the [population](https://statsandr.com/blog/what-is-the-difference-between-population-and-sample/) (this is the reason all areas are gray and not colored in the mosaic plot). Try to increase the sample size, and you will probably reject the null hypothesis.

So to recap, yes a single *p*-value is sufficient in my opinion, and if you want more granularity you can refer to a mosaic or barplot.

Hope this helps.

Regards,
Antoine

**Comment by eclipsqq** [comment]
do you know how to calculate power for fisher's exact test? thx!

**Reply by AntoineSoetewey** [reply]
To calculate power for Fisher's exact test, you will need to use simulation methods because there isn't a direct formula for power calculation like there is for other statistical tests. See the `power.fisher.test` function from the `{statmod}` package: https://www.rdocumentation.org/packages/statmod/versions/1.5.0/topics/power.fisher.test
